src/OptimistAttestationResolver.sol

```diff
@@ -76,54 +73,47 @@ contract OptimistAttestationResolver is
+        require(AllowlistResolverUpgradeable.onAttest(attestation, value), "OptimistAttestationResolver: attester is not allowed");
+        require(attestationUidByRecipient[attestationInput.recipient] == bytes32(0), "OptimistAttestationResolver: recipient already has an attestation");
         attestationUidByRecipient[attestationInput.recipient] = attestationInput.uid;
-        optimist.mint(attestationInput.recipient);
+        emit AttestationCreated(attestationInput.recipient);
         return true;
     }
 
     /// @inheritdoc SchemaResolverUpgradeable
     function onRevoke(
         Attestation calldata attestationInput,
-        uint256
+        uint256 value
     ) internal whenNotPaused override(SchemaResolverUpgradeable, AllowlistResolverUpgradeable) returns (bool) {
-        require(!allowedAttesters[attestationInput.attester], "OptimistAttestationResolver: attester is not allowed");
-        require(attestationUidByRecipient[attestationInput.recipient] != bytes32(0), "OptimistAttestationResolver: recipient not record by uid");
+        require(AllowlistResolverUpgradeable.onRevoke(attestation, value), "OptimistAttestationResolver: attester is not allowed");
+        require(attestationUidByRecipient[attestationInput.recipient] != bytes32(0), "OptimistAttestationResolver: recipient does not have an attestation");
         attestationUidByRecipient[attestationInput.recipient] = bytes32(0);
+        emit AttestationRevoked(attestationInput.recipient);
         return true;
     }
 
-    /**
-    * @dev Pause the contract or not.
-    * @param enableOrNot A flag used to determine whether to pause.
-    */
-    function enablePaused(bool enableOrNot) external onlyRole(PAUSE_ROLE) {
-        if (enableOrNot) {
-            _pause();
-        } else {
-            _unpause();
-        }
+    function pause() external onlyRole(PAUSER_ROLE) {
+        _pause();
     }
 
-    /**
-    * @dev Allow or remove attester.
-    * @param enableOrNot A flag used to determine whether to allow or remove.
-    * @param attester The attester address
-    */
-    function enableAllowAttester(bool enableOrNot, address attester) external onlyRole(ALLOWLIST_ROLE) {
-        if (enableOrNot) {
-            _allowAttester(attester);
-            emit OptimistAttestationCreated(attester);
-        } else {
-            _removeAttester(attester);
-            emit OptimistAttestationRevoked(attester);
-        }
+    function unpause() external onlyRole(PAUSER_ROLE) {
+        _unpause();
     }
+
+    function addAttesterToAttesterAllowlist(address _newAttester) external onlyRole(ADMIN_ROLE) {
+        _allowAttester(_newAttester);
+        emit AttesterAdded(_newAttester);
+    }
+
+    function removeAttesterFromAttesterAllowlist(address _attesterToRemove) external onlyRole(ADMIN_ROLE) {
+        _removeAttester(_attesterToRemove);
+        emit AttesterRemoved(_attesterToRemove);
+    }
```

Comment:
1. split out add/remove operation into separate functions
2. I think only ADMIN_ROLE should be allowed to add/remove from the allowlist attesters? is that the intended behavior?

Reply:
no, buy I think ADMIN_ROLE is way more powerful, just add an ALLOWLIST_ROLE to add/remove from the allowlist attesters